File: api/api/Models/Leg.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Models
{
    [Table("Legs")]
    public class Leg
    {
        public int LegId { get; set; }
        public int TripId { get; set; }
        public Trip Trip { get; set; }
        public string DepartureAirportId { get; set; }
        public string ArrivalAirportId { get; set; }
        public DateTime DepartureDate { get; set; }
        public DateTime ArrivalDate { get; set;}
        public double DistanceMiles { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: frontend/src/Models/Leg.ts
export type LegGet = {
    legId: number;
    tripId: number;
    departureAirportId: string;
    arrivalAirportId: string;
    departureDate: string;
    arrivalDate: string;
    distanceMiles: number;
    createdAt: string;
}

export type LegPost = {
    tripId: number;
    departureAirportId: string;
    arrivalAirportId: string;
    departureDate: string;
    arrivalDate: string;
}

export type LegDelete = {
    id: number;
}